src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -1886,89 +1901,81 @@ private static void validateChapterWiseTesting(Iterator<Node> itrChecks,
             Node config = null;
 
             try {
-                config = itrConfigs.next();
+                config = itrChecks.next();
```

Comment:
I don't understand why we rely on `NoSuchElementException` when there is no next element instead of checking it via `itrChecks.hasNext()`. However, this old logic is not related to changes in this PR.

Reply:
Agreed, an `if` and `hasNext` is better. This is old code and I didn't write it well all those years ago.
ffaf99ed9905b6df24bdeb770bfee43066db7b75

Reply:
@Zopsss , as you need to change this PR, please recheck is we can improve this code by hasNext.

Reply:
done

https://github.com/checkstyle/checkstyle/blob/cc4909e7907de21f6026e5c703cf9e4e128468a9/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java#L1902-L1906